### PR TITLE
feat: add load_extensions method

### PR DESCRIPTION
**Pull request by maksims8**
## Pull Request Type

- [X] Feature addition
- [ ] Bugfix
- [ ] Documentation update
- [ ] Code refactor
- [ ] Tests improvement
- [ ] CI/CD pipeline enhancement
- [ ] Other: [Replace with a description]

## Description
This commit introduces a new method in which you can overcome manually loading extensions you have in your project down to just one line and therefore allows your project setup time and complexity to be decreased.

## Changes

* Introduces a new method to load extensions without manually looping and loading.

## Test Scenarios
`Non-recursive (loaded exts.example):`
```py
import interactions

intents = interactions.Intents.DEFAULT | interactions.Intents.MESSAGE_CONTENT
client = interactions.Client(intents=intents)

@interactions.listen()
async def on_ready():
    print("Client is ready.")

client.load_extensions("exts") # Folder is located in the root of the project
client.start("TOKEN")
```

`Recursive (loaded exts.example & exts.subdir.example_two):`
```py
import interactions

intents = interactions.Intents.DEFAULT | interactions.Intents.MESSAGE_CONTENT
client = interactions.Client(intents=intents)

@interactions.listen()
async def on_ready():
    print("Client is ready.")

client.load_extensions("exts", recursive=True) # Folder is located in the root of the project
client.start("TOKEN")
```

## Python Compatibility

- [X] I've ensured my code works on Python `3.10.x`
- [X] I've ensured my code works on Python `3.11.x`

## Checklist

- [X] I've run the `pre-commit` code linter over all edited files
- [X] I've tested my changes on supported Python versions
- [X] I've added tests for my code, if applicable
- [X] I've updated / added documentation, where applicable

### CODE REVIEW

File: interactions/client/client.py [hunk 1]
```diff
@@ -2002,6 +2004,31 @@ def load_extension(
         module = importlib.import_module(module_name, package)
         self.__load_module(module, module_name, **load_kwargs)
 
+    def load_extensions(
+        self,
+        package: str,
```
Comment (i0bs) [inline interactions/client/client.py]: Can we turn this into `*args`?

**Review by i0bs — CHANGES_REQUESTED** [review]

File: interactions/client/client.py (hunk 1 above)
Comment (maksims8) [reply interactions/client/client.py]: Now that you mention that, I was first planning to use `*packages` but I completely forgot. I will implement this within half an hour.

**Review by i0bs — APPROVED** [review]

File: interactions/client/client.py [hunk 2]
```diff
@@ -2002,6 +2004,36 @@ def load_extension(
...
+        for package in packages:
+            # If recursive then include subdirectories ('**')
+            # otherwise just the package specified by the user.
+            pattern = os.path.join(package, "**" if recursive else "", "*.py")
+
+            # Find all files matching the pattern, and convert slashes to dots.
+            extensions = [f.replace(os.path.sep, ".").replace(".py", "") for f in glob.glob(pattern, recursive=True)]
+
+            for ext in extensions:
+                print(ext)
```
Comment (silasary) [inline interactions/client/client.py]: Either remove this, or change it to a nicer message sent through the logger.

**Review by silasary — CHANGES_REQUESTED** [review]
Other than a stray print() statement, this looks really good

File: interactions/client/client.py (hunk 2 above)
Comment (maksims8) [reply interactions/client/client.py]: My testing print statement, thanks for pointing that out.

**Review by silasary — APPROVED** [review]

**Pull request merged**